src/pages/safe/apps.tsx

```diff
@@ -1,18 +1,33 @@
 import type { NextPage } from 'next'
 import Head from 'next/head'
+import { useRouter } from 'next/router'
 import Grid from '@mui/material/Grid'
 import { useSafeApps } from '@/hooks/safe-apps/useSafeApps'
 import { AppCard } from '@/components/safe-apps/AppCard'
 import { SafeAppsHeader } from '@/components/safe-apps/SafeAppsHeader'
+import AppFrame from '@/components/safe-apps/AppFrame'
+import SafeAppsErrorBoundary from '@/components/safe-apps/SafeAppsErrorBoundary'
+import SafeAppsLoadError from '@/components/safe-apps/SafeAppsErrorBoundary/SafeAppsLoadError'
 import { IS_PRODUCTION } from '@/config/constants'
 
 const Apps: NextPage = () => {
+  const router = useRouter()
+  const { appUrl } = router.query
+
   const { allSafeApps, remoteSafeAppsLoading, customSafeAppsLoading, addCustomApp } = useSafeApps()
 
   if (remoteSafeAppsLoading || customSafeAppsLoading) {
     return <p>Loading...</p>
   }
 
+  if (appUrl) {
+    return (
+      <SafeAppsErrorBoundary render={() => <SafeAppsLoadError />}>
+        <AppFrame appUrl={appUrl as string} />
```

Comment:
Instead of casting we should do smth like `if (Array.isArray(appUrl)) appUrl = appUrl[0]`.

Reply:
done

Reply:
Thanks! LGTM 👍